### PR TITLE
<TreeView /> - add demo with custom rendering

### PR DESCRIPTION
**Pull request by liorzisman**

### CODE REVIEW

File: demo/components/tree-view-demo.tsx [hunk 1]
```diff
@@ -37,15 +40,62 @@ function SelectedItem({selectedItem}: any) {
 export interface TreeViewDemoState {
     selectedItem: TreeItemData | undefined;
     focusedItem: TreeItemData | undefined;
+    selectedItemCustom: TreeItemData | undefined;
+    focusedItemCustom: TreeItemData | undefined;
 }
 
+export const CustomItem: React.SFC<TreeItemProps> =
```
Comment (liorzisman) [inline demo/components/tree-view-demo.tsx]: Will further improve the custom rendering

Comment (JoMarton) [inline demo/components/tree-view-demo.tsx]: This demo is non relatable. Lets think of a more real world example?

File: demo/components/tree-view-demo.tsx [hunk 2]
```diff
@@ -64,6 +114,19 @@ export class TreeViewDemo extends React.Component<{}, TreeViewDemoState> {
                         selectedItem={this.state.selectedItem}
                     />
                 </section>
+                <h3>TreeView with custom item renderer</h3>
+                <section>
+                    <SelectedItem selectedItem={this.state.selectedItemCustom} />
+                    <br />
+                    <TreeView
```
Comment (JoMarton) [inline demo/components/tree-view-demo.tsx]: Please separate your demo to two components

Comment (JoMarton) [inline demo/components/tree-view-demo.tsx]: Also please add a relevant usability test to this demo

File: demo/components/tree-view-demo.tsx (hunk 1 above)
Comment (liorzisman) [reply demo/components/tree-view-demo.tsx]: a food menu is relatable in my taste, especially the desserts part :)

File: demo/components/tree-view-demo.tsx (hunk 2 above)
Comment (liorzisman) [reply demo/components/tree-view-demo.tsx]: why separate? this component thing represents the demo for the <TreeView /> as a whole.
will test it as you suggested

**Pull request merged**